app/upgrades/v17/upgrades.go

```diff
@@ -0,0 +1,327 @@
+		// (these will get set in the OnChanAck callback)
+		// create community pool deposit account
+		connectionId := hostZone.ConnectionId
+		connectionEnd, found := k.IBCKeeper.ConnectionKeeper.GetConnection(ctx, connectionId)
+		if !found {
+			return errorsmod.Wrapf(connectiontypes.ErrConnectionNotFound, "connection %s not found", connectionId)
+		}
+		counterpartyConnectionId := connectionEnd.Counterparty.ConnectionId
+
+		appVersion := string(icatypes.ModuleCdc.MustMarshalJSON(&icatypes.Metadata{
+			Version:                icatypes.Version,
+			ControllerConnectionId: connectionId,
+			HostConnectionId:       counterpartyConnectionId,
+			Encoding:               icatypes.EncodingProtobuf,
+			TxType:                 icatypes.TxTypeSDKMultiMsg,
+		}))
+
+		depositAccount := stakeibctypes.FormatHostZoneICAOwner(chainId, stakeibctypes.ICAAccountType_COMMUNITY_POOL_DEPOSIT)
+		if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, connectionId, depositAccount, appVersion); err != nil {
+			return errorsmod.Wrapf(stakeibctypes.ErrFailedToRegisterHostZone, "failed to register community pool deposit ICA")
+		}
+
+		returnAccount := stakeibctypes.FormatHostZoneICAOwner(chainId, stakeibctypes.ICAAccountType_COMMUNITY_POOL_RETURN)
+		if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, connectionId, returnAccount, appVersion); err != nil {
+			return errorsmod.Wrapf(stakeibctypes.ErrFailedToRegisterHostZone, "failed to register community pool return ICA")
+		}
+	}
+
+	return nil
+}
+
+// Deletes all stale queries
+func DeleteAllStaleQueries(ctx sdk.Context, k icqkeeper.Keeper) {
+	for _, query := range k.AllQueries(ctx) {
+		if query.CallbackId == stakeibckeeper.ICQCallbackID_Delegation {
+			k.DeleteQuery(ctx, query.Id)
+		}
+	}
+}
+
+// Resets the slash query in progress flag for each validator
+func ResetSlashQueryInProgress(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+		for i, validator := range hostZone.Validators {
+			validator.SlashQueryInProgress = false
+			hostZone.Validators[i] = validator
+		}
+		k.SetHostZone(ctx, hostZone)
+	}
+}
+
+// Increases the community pool tax from 2 to 5%
+// This was from prop 223 which passed, but was deleted due to an ICS blacklist
+func IncreaseCommunityPoolTax(ctx sdk.Context, k distributionkeeper.Keeper) error {
+	params := k.GetParams(ctx)
+	params.CommunityTax = CommunityPoolTax
+	return k.SetParams(ctx, params)
+}
+
+// Updates the outer redemption rate bounds
+func UpdateRedemptionRateBounds(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+		// Give osmosis a bit more slack since OSMO stakers collect real yield
+		outerAdjustment := RedemptionRateOuterMaxAdjustment
+		if hostZone.ChainId == OsmosisChainId {
+			outerAdjustment = outerAdjustment.Add(OsmosisRedemptionRateBuffer)
+		}
+
+		outerMinDelta := hostZone.RedemptionRate.Mul(RedemptionRateOuterMinAdjustment)
```

Comment:
nit: would this be simpler?
```
outerMin := hostZone.RedemptionRate.Mul(1-RedemptionRateOuterMinAdjustment)
```

Reply:
yeah probably, but would prefer to not change it at this point

Reply:
ok